.github/workflows/qiita-ci.yml

```diff
@@ -0,0 +1,207 @@
+        shell: bash -l {0}
+        run: |
+          echo "Testing: " $COVER_PACKAGE
+
+          # pull out the port so we can modify the configuration file easily
+          pgport=${{ job.services.postgres.ports[5432] }}
+          sed -i "s/PORT = 5432/PORT = $pgport/" qiita_core/support_files/config_test.cfg
+
+          # PGPASSWORD is read by pg_restore, which is called by the build_db process.
+          export PGPASSWORD=postgres
+
+          # Setting up main qiita conda environment
+          conda config --add channels conda-forge
+          conda create -q --yes -n qiita python=3.6 pip libgfortran numpy nginx cython redis
+          conda activate qiita
+          pip install sphinx sphinx-bootstrap-theme nose-timer codecov Click
+
+          # Configuring SSH
+          cp /etc/ssh/sshd_config sshd_config
+          echo "RSAAuthentication yes" > sshd_config
+          echo "PubkeyAuthentication yes" > sshd_config
+          echo "StrictModes no" > sshd_config
+          sudo mv sshd_config /etc/ssh/sshd_config
+          sudo systemctl restart ssh
+          ps -elf | grep ssh
+
+      - name: Webdis install
+        shell: bash -l {0}
+        run: |
+          sudo apt-get -y install libevent-dev
+          git clone https://github.com/nicolasff/webdis
+          cd webdis
+          make
+
+      - name: Main install
+        shell: bash -l {0}
+        run: |
+          conda activate qiita
+          export QIITA_SERVER_CERT=`pwd`/qiita_core/support_files/server.crt
+          export QIITA_CONFIG_FP=`pwd`/qiita_core/support_files/config_test.cfg
+          export REDBIOM_HOST="http://localhost:7379"
+          pip install . --no-binary redbiom
+          pwd
+          mkdir ~/.qiita_plugins
+
+      - name: Install plugins
+        shell: bash -l {0}
+        run: |
+          wget https://data.qiime2.org/distro/core/qiime2-2019.4-py36-linux-conda.yml
+          conda env create -q -n qtp-biom --file qiime2-2019.4-py36-linux-conda.yml
+          rm qiime2-2019.4-py36-linux-conda.yml
+          export QIITA_SERVER_CERT=`pwd`/qiita_core/support_files/server.crt
+          export QIITA_CONFIG_FP=`pwd`/qiita_core/support_files/config_test.cfg
+          export REDBIOM_HOST="http://localhost:7379"
+          conda activate qtp-biom
+          pip install -U pip
+          pip install https://github.com/qiita-spots/qiita_client/archive/master.zip
+          pip install https://github.com/qiita-spots/qtp-biom/archive/master.zip
+          configure_biom --env-script "source /home/runner/.profile; conda activate qtp-biom" --server-cert $QIITA_SERVER_CERT
+
+      - name: Starting services
+        shell: bash -l {0}
+        run: |
+          conda activate qiita
+          export QIITA_SERVER_CERT=`pwd`/qiita_core/support_files/server.crt
+          export QIITA_CONFIG_FP=`pwd`/qiita_core/support_files/config_test.cfg
+          export REDBIOM_HOST="http://localhost:7379"
+
+          echo "1. Setting up redis"
```

Comment:
Instead of using echo, another alternative is to put each of these sections under a `-name: Setting up redis` tag so that the UI summarizes which step failed, etc.

Reply:
I'm OK with echos! :)